server/node.go

```diff
@@ -235,8 +235,11 @@ func NewNode(ctx storage.StoreContext, recorder *status.MetricsRecorder, stopper
 }
 
 // context returns a context encapsulating the NodeID.
-func (n *Node) context() context.Context {
-	return log.Add(context.Background(), log.NodeID, n.Descriptor.NodeID)
+func (n *Node) context(ctx context.Context) context.Context {
+	if ctx == nil {
+		ctx = context.Background()
```

Comment:
I had it the other way, but I took a note from [storage.Store](https://github.com/cockroachdb/cockroach/blob/master/storage/store.go#L624).

We should decide on how we want to do this type of logic, because we run into the same question with `replica.context`.